### PR TITLE
Implemented fix for missing final date value

### PR DESCRIPTION
**Pull request by Ro4052**
I need some more test cases for this fix. I'm not confident that it will work in all cases.

### CODE REVIEW

**Comment by Ro4052** [comment]
I've confirmed that this only works in the specific case that I was testing. I'm not sure how to approach this since when we calculate the extents we don't know how much we would need to pad the right.

**Comment by Ro4052** [comment]
This new approach seems to work but it looks a bit messy to me.

**Comment by andy-lee-eng** [comment]
I think we also need to change the bar alignment when not using ordinal scales (with padding). Please see the changes I've added. This also includes splitting the padding so it's on both sides.

**Comment by Ro4052** [comment]
That definitely looks better. I'm looking into another way of calculating the padding that should hopefully bit a little nicer.

**Review by andy-lee-eng — APPROVED** [review]

File: packages/perspective-viewer-d3fc/src/js/axis/crossAxis.js [hunk 1]
```diff
@@ -46,13 +46,19 @@ export const domain = settings => {
...
+        .accessors([d => new Date(d[valueName])])
+        .padUnit("domain");
 
     const _domain = function(data) {
         const flattenedData = flattenArray(data);
         switch (axisType(settings, settingName)) {
             case AXIS_TYPES.time:
-                return extentTime(flattenedData);
+                const dataStart = flattenedData[0][valueName];
+                const dataNext = flattenedData.find(d => d[valueName] !== dataStart);
```
Comment (Ro4052) [inline packages/perspective-viewer-d3fc/src/js/axis/crossAxis.js]: @DevAndyLee, can you think of a better way to do this? I.e. find the date value of the next bar. This doesn't feel like a very robust method, unless we rely on the data always being ordered correctly.

**Comment by Ro4052** [comment]
I've hopefully switched to a more robust implementation, may need some variable changing for clarity.

**Review by andy-lee-eng — APPROVED** [review]

**Pull request merged**